### PR TITLE
Fix Subject Counts when Course has multiple course_runs

### PR DESCRIPTION
**Pull request by burhandodhy**
**Description**
This PR fixes the incorrect subject counts when the course has multiple course runs.

**Jira Ticket**
https://edlyio.atlassian.net/browse/EDLY-3055

### CODE REVIEW

**Review by taimoor-ahmed-1 — APPROVED** [review]

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ def choices(self):
         return OrderedDict(sorted([(x.slug, x.name) for x in Subject.objects.all()], key=lambda x: x[1]))
 
     def get_number_of_courses(self, obj):
-        return obj.course_set.count()
+        return CourseRun.objects.filter(course__subjects=obj, status=CourseRunStatus.Published).count()
```
Comment (iam-mhaseeb) [inline course_discovery/apps/api/serializers.py:269]: Is it working? If yes, I'm wondering how?? 🤔
In my sight it should be like this:
```
CourseRun.objects.filter(course__subjects__in=obj, status__slug=CourseRunStatus.Published).count()
```

**Review by iam-mhaseeb — CHANGES_REQUESTED** [review]

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (burhandodhy) [reply course_discovery/apps/api/serializers.py:269]: @iam-mhaseeb its working fine 
image

where this query `CourseRun.objects.filter(course__subjects__in=obj).count()`  return error `TypeError: 'Subject' object is not iterable`

Comment (iam-mhaseeb) [reply course_discovery/apps/api/serializers.py:269]: There could be multiple subjects attached to one course, this is actually concerning me and we are just checking a single object.

**Review by iam-mhaseeb — APPROVED** [review]

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍

**Pull request merged**